packages/yourloops/components/device/parameters-change-history.tsx

```diff
@@ -59,17 +59,18 @@ interface ParametersChangeHistoryProps {
 export const ParametersChangeHistory: FC<ParametersChangeHistoryProps> = ({ history, goToDailySpecificDate }) => {
   const theme = useTheme()
   const { t } = useTranslation()
+  const [historyToDisplay, setHistoryToDisplay] = useState<ParametersChange[]>([])
 
   const onClickChangeDate = (date: number): void => {
     goToDailySpecificDate(date)
     window.scroll(0, 0)
   }
 
   useEffect(() => {
-    sortHistoryParametersByDate(history).reverse()
-    sortPumpSettingsParameterByLevel(history)
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+    const historyParametersSortedByDate = sortHistoryParametersByDate(history).reverse()
+    const historyPumpSettingSortedByLevel = sortPumpSettingsParameterByLevel(historyParametersSortedByDate)
+    setHistoryToDisplay(historyPumpSettingSortedByLevel)
+  }, [history])
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏For consistency's sake and to avoid any side effects, maybe we should also use `historyToDisplay` instead of `history` line 140, when passing data to `ParameterChangeValue`?

(I cannot put the comment on the related line, sorry 😬)